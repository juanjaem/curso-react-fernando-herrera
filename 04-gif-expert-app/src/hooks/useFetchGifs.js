import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGifs';
;

export const useFectchGifs = (category) => {
    const [state, setstate] = useState({
        data: [],
        loading: true
    })
    console.log('11111');
    useEffect(() => {
        console.log('22222');
        getGifs(category).then(imgs =>
            setstate({ data: imgs, loading: false })
        );
    }, [category])

    return state;
}
