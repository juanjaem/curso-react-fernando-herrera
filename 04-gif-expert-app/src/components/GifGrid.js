import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { useFectchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
    const { data: images, loading } = useFectchGifs(category);

    return (
        <Fragment>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>

            { loading && <p className="animate__animated animate__flash">Cargando...</p> }

            <div className="card-grid">
                {
                    images.map(img => (<GifGridItem key={ img.id } {...img} />))
                }
            </div>
        </Fragment>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
