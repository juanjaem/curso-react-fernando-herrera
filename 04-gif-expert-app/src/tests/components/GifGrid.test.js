import { shallow } from 'enzyme';
import '@testing-library/jest-dom'; // para el autocompletado
import { GifGrid } from '../../components/GifGrid';
import React from 'react';
import { useFectchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('pruebas en <GifGrid />', () => {
    const category = 'prueba';

    test('debe de mostrar el componente correctamente', () => {
        useFectchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
        const gifs = [
            {
                id: 'ABC',
                url: 'https://loquesea.gif',
                title: 'cualquier cosa'
            }
        ];

        useFectchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBeFalsy();
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
})
