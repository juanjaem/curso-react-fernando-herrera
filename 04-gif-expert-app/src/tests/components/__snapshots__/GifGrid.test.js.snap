// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGrid /> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    prueba
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`pruebas en <GifGrid /> debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    prueba
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="https://loquesea.gif"
    />
  </div>
</Fragment>
`;
